### PR TITLE
userlib: Move 32-bit struct members to end of struct

### PR DESCRIPTION
**Pull request by furquan-rivos**
This change moves 32-bit struct members in syscall stack args structure to end of struct to ensure that the 64-bit (usize) struct members are properly aligned on a 64-bit system. ASM code in arm_m.rs and riscv32.rs is accordingly updated.

TEST=Ensured that rv32 still boots and tests pass for hifive-inventor on qemu.

### CODE REVIEW

File: sys/userlib/src/arch/riscv32.rs [hunk 1]
```diff
@@ -225,11 +225,11 @@ pub(crate) unsafe extern "C" fn sys_borrow_write_stub(
...
-        lw a4, 4*4(a0)
-        lw a3, 3*4(a0)
-        lw a2, 2*4(a0)
-        lw a1, 1*4(a0)
-        lw a0, 0*4(a0)
+        lw a4, 3*4(a0)
+        lw a3, 2*4(a0)
+        lw a2, 1*4(a0)
+        lw a1, 0*4(a0)
+        lw a0, 4*4(a0)
```
Comment (adurbin-rivos) [inline sys/userlib/src/arch/riscv32.rs:232]: Do we really need to change the assembly? I don't see the kernel side of this, but if it's just taking parameters out and putting them in memory does the order matter for the purposes of the assembly call?

Comment (furquan-rivos) [reply sys/userlib/src/arch/riscv32.rs:232]: Yes, the change in assembly is required. The reason is as follows:

The structures that are being changed (*Args) are not for the user<->kernel interface. These are used within `sys/userlib` across the arch-agnostic <-> arch-dependent code boundary. Example: [`SendArgs`](https://github.com/rivosinc/hubris/blob/rivos/main-rv64/sys/userlib/src/lib.rs#L148). This is populated by [`sys_send`](https://github.com/rivosinc/hubris/blob/rivos/main-rv64/sys/userlib/src/lib.rs#L126) which is arch-agnostic code and used by [`sys_send_stub`](https://github.com/rivosinc/hubris/blob/rivos/main-rv64/sys/userlib/src/arch/riscv32.rs#L77) which is in arch-dependent code. The arch dependent code is expected to pick these parameters off the stack where the struct members are populated in registers as expected by the user<->kernel interface.

Comment (adurbin-rivos) [reply sys/userlib/src/arch/riscv32.rs:232]: Got it.

**Review by adurbin-rivos — APPROVED** [review]

**Review by zachreizner — APPROVED** [review]

**Review by FawazTirmizi — APPROVED** [review]

**Review by beezow — APPROVED** [review]
This may cause some issues with Humility and ARM targets. It makes lots of assumptions about the order and how things are pushed when doing syscalls and decoding syscall arguments.  I don't think we need to stall this PR though, humility is probably already broken with this branch.
I created a [jira ticket](https://rivosinc.atlassian.net/browse/SW-557) to track this.

**Comment by furquan-rivos** [comment]
> This may cause some issues with Humility and ARM targets. It makes lots of assumptions about the order and how things are pushed when doing syscalls and decoding syscall arguments. I don't think we need to stall this PR though, humility is probably already broken with this branch.
> I created a [jira ticket](https://rivosinc.atlassian.net/browse/SW-557) to track this.

Thanks for catching this, Drew! I believe this will impact both ARM and RV32?

**Comment by furquan-rivos** [comment]
> > This may cause some issues with Humility and ARM targets. It makes lots of assumptions about the order and how things are pushed when doing syscalls and decoding syscall arguments. I don't think we need to stall this PR though, humility is probably already broken with this branch.
> > I created a [jira ticket](https://rivosinc.atlassian.net/browse/SW-557) to track this.
> 
> Thanks for catching this, Drew! I believe this will impact both ARM and RV32?

One more thing - the changes in this PR impact the interface between arch-agnostic and arch-dependent code. However, there is no change across the user/kernel boundary. My assumption was that humility would be looking at the argument registers and not the stack. But, I haven't looked at the humility code to confirm this.

**Pull request merged**